"odo config set" doesn't validate port values and thus "odo describe" panics

/kind bug

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the Google group if you have a question rather than a bug or feature request.

The group is at: https://groups.google.com/forum/#!forum/odo-users

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?

**Operating System:** all

**Output of `odo version`:** master

## How did you run odo exactly?
https://github.com/openshift/odo/pull/3008#discussion_r416561531
```sh
$ odo config set ports 909,808 -f
 ✓  Local config successfully updated

Run `odo push --config` command to apply changes to the cluster

$ odo describe 
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/openshift/odo/pkg/component.getPortsProtocolMapping(0xc0006136c0, 0x2, 0x2, 0x0)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/component/component.go:991 +0x177
github.com/openshift/odo/pkg/component.GetComponentFromConfig(0xc000ba7500, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/mrinaldas/go/src/github.com/openshift/odo/pkg/component/component.go:962 +0x6d6
github.
```

## Actual behavior
`odo describe` panics because `odo config set` doesn't perform a check for ports provided by the user

## Expected behavior
odo should validate that provided value for ports follow `<port>/<protocol>` format and the values are valid.

## Any logs, error output, etc?

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -200,7 +200,7 @@ func newProxyLocalConfig() proxyLocalConfig {
 func (lci *LocalConfigInfo) SetConfiguration(parameter string, value interface{}) (err error) {
 	// getting the second arg makes sure that this never panics
 	strValue, _ := value.(string)
-	if parameter, ok := asLocallySupportedParameter(parameter); ok {
+	if parameter, ok := AsLocallySupportedParameter(parameter); ok {
 		switch parameter {
 		case "type":
 			lci.componentSettings.Type = &strValue
@@ -326,7 +326,7 @@ func (lci *LocalConfigInfo) ConfigFileExists() bool {
 
 // DeleteConfiguration is used to delete config from local odo config
 func (lci *LocalConfigInfo) DeleteConfiguration(parameter string) error {
-	if parameter, ok := asLocallySupportedParameter(parameter); ok {
+	if parameter, ok := AsLocallySupportedParameter(parameter); ok {
 
 		switch parameter {
 		case "cpu":
@@ -362,7 +362,7 @@ func (lci *LocalConfigInfo) DeleteURL(parameter string) error {
 // parameter is the name of the config parameter
 // value is the value to be deleted
 func (lci *LocalConfigInfo) DeleteFromConfigurationList(parameter string, value string) error {
-	if parameter, ok := asLocallySupportedParameter(parameter); ok {
+	if parameter, ok := AsLocallySupportedParameter(parameter); ok {
 		switch parameter {
 		case "storage":
 			for i, storage := range lci.GetStorage() {
@@ -653,7 +653,8 @@ func FormatLocallySupportedParameters() (result string) {
 	return "\nAvailable Local Parameters:\n" + result
 }
 
-func asLocallySupportedParameter(param string) (string, bool) {
+// AsLocallySupportedParameter returns the parameter in lower case and a boolean indicating if it is a supported parameter
+func AsLocallySupportedParameter(param string) (string, bool) {
 	lower := strings.ToLower(param)
 	return lower, lowerCaseLocalParameters[lower]
 }

--- a/pkg/odo/cli/config/set.go
+++ b/pkg/odo/cli/config/set.go
@@ -9,6 +9,7 @@ import (
 	clicomponent "github.com/openshift/odo/pkg/odo/cli/component"
 	"github.com/openshift/odo/pkg/odo/cli/ui"
 	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/odo/util/validation"
 	"github.com/pkg/errors"
 
 	"github.com/spf13/cobra"
@@ -62,6 +63,7 @@ func (o *SetOptions) Complete(name string, cmd *cobra.Command, args []string) (e
 		o.paramName = args[0]
 		o.paramValue = args[1]
 	}
+
 	// we initialize the context irrespective of --now flag being provided
 	if o.now {
 		o.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
@@ -152,6 +154,33 @@ func (o *SetOptions) Run() (err error) {
 	return nil
 }
 
+func isValidArgumentList(args []string) error {
+
+	if len(args) < 2 {
+		return fmt.Errorf("please provide a parameter name and value")
+	} else if len(args) > 2 {
+		return fmt.Errorf("only one value per parameter is allowed")
+	}
+
+	var err error
+	param, ok := config.AsLocallySupportedParameter(args[0])
+
+	if !ok {
+		err = errors.Errorf("the provided parameter is not supported, %v", args[0])
+	}
+
+	switch param {
+	case "ports", "debugport":
+		err = validation.PortsValidator(args[1])
+	}
+
+	if err != nil {
+		err = errors.Errorf("validation failed for the provided arguments, %v", err)
+	}
+
+	return err
+}
+
 // NewCmdSet implements the config set odo command
 func NewCmdSet(name, fullName string) *cobra.Command {
 	o := NewSetOptions()
@@ -169,15 +198,7 @@ func NewCmdSet(name, fullName string) *cobra.Command {
 				}
 				return nil
 			}
-
-			if len(args) < 2 {
-				return fmt.Errorf("please provide a parameter name and value")
-			} else if len(args) > 2 {
-				return fmt.Errorf("only one value per parameter is allowed")
-			} else {
-				return nil
-			}
-
+			return isValidArgumentList(args)
 		}, Run: func(cmd *cobra.Command, args []string) {
 			genericclioptions.GenericRun(o, cmd, args)
 		},

